bin/osg-topology

```diff
@@ -166,48 +166,148 @@ def list_security_contacts(args):
-                    for contact in security_contacts:
-                        key = None
-                        if 'Name' in contact and 'Email' in contact:
-                            key = "%s<%s>" % (contact['Name'], contact['Email'])
-                        elif 'Name' in contact:
-                            key = contact['Name']
-                        if key:
-                            keys.add(key)
-
-    sites = results.keys()
-    sites.sort()
-    for site in sites:
-        keys = list(results[site])
-        keys.sort()
-        print("- Resource: %s" % site)
-        for key in keys:
-             print(u"  * %s" % key)
+
+                if name and contact_list_info:
+                    results[name] = contact_list_info
+
+    print_contacts(args, 'Resource', results)
+
+def list_vo_contacts(args):
+    """
+    List vo contacts for OSG.
+    """
+    base_url = "https://my.opensciencegrid.org/vosummary/xml" \
+                "?summary_attrs_showcontact=on&all_vos=on" \
+                "&active_value=1&active=on&disable=on&disable_value=0"
+
+    with get_auth_session(args) as session:
+        url = mangle_url(base_url, args, session)
+        #print(url)
+        response = session.get(url)
+
+    if response.status_code != requests.codes.ok:
+        print("MyOSG request failed (status %d): %s" % (response.status_code,
+              response.text[:2048]), file=sys.stderr)
+        return 1
+
+    root = ET.fromstring(response.content)
+    if root.tag != 'VOSummary':
+        print("MyOSG returned invalid XML with root tag %s" % root.tag,
+              file=sys.stderr)
+        return 1
+
+    results = {}
+    for child_vo in root:
+        if child_vo.tag != "VO":
+            print("MyOSG returned a non-VO (%s) inside summary." % \
+                  root.tag, file=sys.stderr)
+            return 1
+        name = None
+        contact_list_info = []
+        for item in child_vo:
+            if item.tag == 'Name':
+                name = item.text
+                continue
+            if item.tag != "ContactTypes":
+                continue
+            for contact_type in item:
+                if contact_type.tag == 'ContactType':
+                    for child in contact_type:
+                        if child.tag == 'Type':
+                            contact_list_type = child.text.lower()
+                        if child.tag != 'Contacts':
+                            continue
+                        for contact in child:
+                            contact_info = { 'ContactType' : contact_list_type }
+                            for contact_contents in contact:
+                                contact_info[contact_contents.tag] = \
+                                    contact_contents.text
+                            contact_list_info.append(contact_info)
+
+        if name and contact_list_info:
+            results[name] = contact_list_info
```

Comment:
Also, once we're in a VO or Resource tag, getting the contact info seems to be exactly the same. Let's refactor that, too

Reply:
Ok, I refactored that too.